src/miral/display_configuration_option.cpp

```diff
@@ -119,12 +123,47 @@ void ScaleSetter::apply_to(mg::DisplayConfiguration& conf)
         });
 }
 
+class AutoscaleSetter : public mg::DisplayConfigurationPolicy
+{
+public:
+    AutoscaleSetter(std::shared_ptr<mg::DisplayConfigurationPolicy> const& base_policy, int target)
+        : base_policy{base_policy},
+          target{target}
+    {
+    }
+
+    void apply_to(mg::DisplayConfiguration& conf) override;
+private:
+    void apply_to(mg::UserDisplayConfigurationOutput& output);
+    std::shared_ptr<mg::DisplayConfigurationPolicy> const base_policy;
+    int target;
+};
+
+void AutoscaleSetter::apply_to(mg::DisplayConfiguration& conf)
+{
+    base_policy->apply_to(conf);
+    conf.for_each_output(
+        [this](mg::UserDisplayConfigurationOutput& output) { if (output.connected) apply_to(output); });
+}
+
+void AutoscaleSetter::apply_to(mg::UserDisplayConfigurationOutput& output)
+{
+    auto const output_height =
+        (output.orientation == mir_orientation_normal || output.orientation == mir_orientation_inverted) ?
+        output.modes[output.current_mode_index].size.height.as_int() :
+        output.modes[output.current_mode_index].size.height.as_int();
+
+    static auto const steps = 4.0;
+    output.scale = roundf((steps * output_height) / target) / steps;
+}
```

Comment:
So this only takes resolution into account… Shouldn't we try and determine whether it's a high pixel density display (from physical size) instead?

Reply:
There are lots of things we *could* take into account, but we don't know what the right choices are.

For example , we could consider screen width (this approach may not be great on tall, thin outputs).

With pixel density we don't know whether the right choice is "Foo fits on screen" or "Foo has readable text".